### PR TITLE
Fix pod state check for jobs

### PR DESCRIPTION
**Pull request by wistefan**
When a job succeeds, the pods are not in state ready. Therefore the pod-readiness check needs to take that into account.
The integration tests checks this through cert-manager, which runs a job on startup.

### CODE REVIEW

File: src/it/cert-manager/src/test/java/io/kokuwa/maven/k3s/PodIT.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package io.kokuwa.maven.k3s;
+
+import org.junit.jupiter.api.Test;
+
+public class PodIT {
+
+	@Test
+	void test() {
+		// the test should only trigger the deployment and assert it does not fail because of the completed job.
```
Comment (sschnabe) [inline src/it/cert-manager/src/test/java/io/kokuwa/maven/k3s/PodIT.java:9]: This can be removed?

**Review by sschnabe — APPROVED** [review]

File: src/it/cert-manager/src/test/java/io/kokuwa/maven/k3s/PodIT.java (hunk 1 above)
Comment (wistefan) [reply src/it/cert-manager/src/test/java/io/kokuwa/maven/k3s/PodIT.java:9]: I thought an empty test is required to actually trigger execution(like in the statefulset it). Comment is only for clarification

Comment (sschnabe) [reply src/it/cert-manager/src/test/java/io/kokuwa/maven/k3s/PodIT.java:9]: If you do not have failsafe configured this class will be ignored anyway.

Comment (wistefan) [reply src/it/cert-manager/src/test/java/io/kokuwa/maven/k3s/PodIT.java:9]: Isnt failsafe inherited from the parent? But I can remove it, no problem.

Comment (sschnabe) [reply src/it/cert-manager/src/test/java/io/kokuwa/maven/k3s/PodIT.java:9]: meh, please ignore my comment

**Pull request merged**